### PR TITLE
doc(graphql): add subscriptions runtime filtering support

### PR DESCRIPTION
**Pull request by sundersc**
_Issue #, if available:_
NA

_Description of changes:_
Add subscriptions runtime filtering support.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/pages/cli/graphql/data-modeling.mdx [hunk 1]
```diff
@@ -453,6 +453,39 @@ type Todo @model {
 If you create a new Todo and don't supply a `content` input, Amplify will ensure that `My new Todo` is auto populated as a value. 
 When `@default` is applied, non-null assertions using `!` are disregarded. For example, `String!` is treated the same as `String`.
 
+## Subscriptions Runtime Filtering
```
Comment (renebrandel) [inline src/pages/cli/graphql/data-modeling.mdx:456]: Server-side filtering for subscriptions

File: src/pages/cli/graphql/data-modeling.mdx [hunk 2]
```diff
@@ -453,6 +453,39 @@ type Todo @model {
...
+
+You can filter the subscriptions server-side by passing a filter expression.
```
Comment (renebrandel) [inline src/pages/cli/graphql/data-modeling.mdx:458]: A server-side subscription filter expression is automatically generated for any @model type.

File: src/pages/cli/graphql/data-modeling.mdx [hunk 3]
```diff
@@ -453,6 +453,39 @@ type Todo @model {
...
+```graphql
+type Task @model {
+  title: String!
+  description: String
+  type: String
+  priority: Int
+}
+```
+
+If you want to subscribe to tasks of type `Security` and priority greater than `5`, you can set the `filter` argument accordingly.
```
Comment (renebrandel) [inline src/pages/cli/graphql/data-modeling.mdx:469]: You can filter the subscriptions server-side by passing a `filter` expression. For example: .....

**Review by renebrandel — APPROVED** [review]

**Review by alharris-at — APPROVED** [review]
Should we leave in some info on the known limitations about limits of how many filters can be applied?

**Review by AaronZyLee — APPROVED** [review]

**Pull request merged**